packages/launchpad/src/components/Input/InlineCodeEditor.vue

```diff
@@ -0,0 +1,59 @@
+<template>
+  <div class="relative text-gray-600 overflow-hidden w-350px">
+    <IconWrapper v-bind="{ ...iconWrapperProps, class: $attrs.class }">
+      <template #default="slotProps">
+        <CodeEditor
+          :class="[slotProps.iconOffsetClasses]"
+          class="font-mono w-full h-full rounded border-transparent disabled:bg-cool-gray-100 disabled:text-cool-gray-400 border-cool-gray-300 focus:border-gray-500 focus:bg-white bg-gray-100 focus:ring-0 focus:outline-none focus:bg-white focus:text-gray-900 border-1 py-2 px-4 whitespace-pre overflow-auto"
+          :readonly="readonly"
+          v-model="localValue"
+        />
+      </template>
+    </IconWrapper>
+  </div>
+</template>
+
+<script lang="ts" setup>
+import { defineProps, defineEmit } from 'vue'
+import CodeEditor from '../CodeEditor.vue'
+import { useModelWrapper } from '../../composables'
+import IconWrapper, { iconProps } from '../IconWrapper.vue'
+import { pick, keys } from 'lodash'
+
+const props = defineProps({
+  modelValue: {
+    type: String,
+  },
+  readonly: {
+    type: Boolean
+  },
+  ...iconProps
+})
+
+const emit = defineEmit(['update:modelValue'])
+const iconWrapperProps = pick(props, keys(iconProps))
```

Comment:
What does this evaluate to? I find it kind of confusing. Maybe a unit test around this (and possibly extract a function with a name that makes it clear what it does?)

Also I think we can just do `v-bind="$attrs"` to achieve the same thing (maybe)?

Reply:
Yeah this is possibly my least favorite line of code. We can't use v-bind $attrs because we already use that to bind $attrs to the native Input API. If we don't "pick" them out, then you just duplicate the $attrs onto the Icons. Does that make sense?